src/main/java/com/twilio/security/RequestValidator.java

```diff
@@ -107,4 +113,33 @@ private boolean secureCompare(String a, String b) {
         return mismatch == 0;
     }
 
+    private String removePort(String url) {
+        try {
+            URI parsedUrl = new URI(url);
+            return (parsedUrl.getPort()) == -1 ? url : url.replaceFirst(":\\d+", "");
```

Comment:
Doesn't work with this: `https://sambino:1password@www.mysite.com:456` (I use very complex passwords)

Maybe add a function like this:
```
private String updatePort(URI url, int newPort) {
    return new URI(
        url.getScheme(),
        url.getUserInfo(),
        url.getHost(), 
        newPort,
        url.getPath(),
        url.getQuery(),
        url.getFragment()).toString();
}
```

Then you can use it for both adding and removing.